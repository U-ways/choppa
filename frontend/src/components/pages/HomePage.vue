<template>
<div class="d-flex justify-content-center align-items-center p-2 h-100">
  <div class="d-flex flex-column align-items-center">
    <ChoppaLogoAtom css="h1" />
    <h1 class="h4">Choppa's Home Page</h1>
    <div>
      <router-link to="/member">Member</router-link> |
      <router-link to="/not-found-404-test">404</router-link>
    </div>
  </div>
</div>
</template>

<script lang="ts">
import ChoppaLogoAtom from "../atoms/ChoppaLogoAtom.vue";
import { Options, Vue } from "vue-class-component";

@Options({
  components: {
    ChoppaLogoAtom
  }
})
export default class HomePage extends Vue {}
</script>

<style scoped lang="scss">
</style>