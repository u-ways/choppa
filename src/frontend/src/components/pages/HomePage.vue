<template>
  <div class="d-flex justify-content-center align-items-center p-2 h-100">
    <div class="d-flex flex-column align-items-center">
      <ChoppaLogoAtom css="h1" />
      <h1 class="h4">Choppa's Home Page</h1>
      <div>
        <router-link to="/tribe/1/">View Tribe Page</router-link> |
        <router-link to="/tribe/1/edit">Edit Tribe Page</router-link> |
        <router-link to="/not-found-404-test">404</router-link>
      </div>
    </div>
  </div>
</template>

<script>
import ChoppaLogoAtom from "@/components/atoms/ChoppaLogoAtom";

export default {
  name: "HomePage",
  components: { ChoppaLogoAtom },
};
</script>

<style scoped>

</style>
