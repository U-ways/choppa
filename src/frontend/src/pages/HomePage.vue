<template>
  <div>
    <section id="hero" class="dark bg-choppa-two pb-6">
      <Navbar/>
      <div class="text-white flex-grow px-3 mt-3 mx-auto lg:mt-6">
        <div class="container mx-auto max-w-screen-lg">
          <div class="flex flex-col lg:flex-row gap-1 place-items-center justify-center">
            <!-- TODO: Change stack to include simplified squad -->
            <img :src="require('@/assets/svg/homepage/stacked-squads.svg')"
                 alt="Stacked squad" height="100%"
                 class="h-72"/>
            <div class="flex-shrink-0 order-first lg:order-last
              text-xl sm:text-2xl xl:text-3xl font-bold text-center">
              <p><span class="lg:inline hidden">Choppa Helps </span>Organise Your Team</p>
              <p>and</p>
              <p>Maximise Knowledge Sharing.</p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section id="organisation">
      <div class="container px-3 text-center mx-auto">
        <div class="row align-items-center text-center pt-6">
          <span class="text-xl sm:text-2xl text-bold">Team Organisation</span> <br/>
          <span class="text-lg sm:text-xl text-semibold">Organise your team structure with our creation tool.</span>
        </div>
        <div>
          <!-- TODO: Place mocked SquadOrganism here. -->
          asdada
        </div>
      </div>
    </section>
    <section id="Knowledge Sharing">
      <div class="">
        <div class="container px-3 text-center mx-auto">
          <div class="row align-items-center text-center pt-6">
            <span class="text-xl sm:text-2xl text-bold">Knowledge Sharing</span> <br/>
            <span class="text-lg sm:text-xl text-semibold">
            Utilise several rotation strategies to <br/>
              maximise knowledge sharing within your team.
          </span>
          </div>
          <div>
            <!-- TODO: Rotation animation here. -->
          </div>
        </div>
      </div>
    </section>
  </div>
</template>

<script>
/* eslint-disable */
import Navbar from "@/components/navbar/Navbar";
import SquadSkeleton from "@/components/squads/SquadSkeleton";

export default {
  name: "HomePage",
  components: {
    SquadSkeleton,
    Navbar,
  },
};
</script>
