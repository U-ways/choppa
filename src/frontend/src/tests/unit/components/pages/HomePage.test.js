import { createLocalVue, describe, expect, it, shallowMount } from "@/tests/unit/test-imports";
import HomePage from "@/components/pages/HomePage";
import VueRouter from "vue-router";

describe("Home Page test", () => {
  it("Renders correctly", async () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();

    const wrapper = shallowMount(HomePage, {
      localVue,
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
