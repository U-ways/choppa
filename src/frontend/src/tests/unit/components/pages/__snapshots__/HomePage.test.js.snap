// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page test Renders correctly 1`] = `
"<div class=\\"d-flex justify-content-center align-items-center p-2 h-100\\">
  <div class=\\"d-flex flex-column align-items-center\\">
    <choppalogoatom-stub css=\\"h1\\"></choppalogoatom-stub>
    <h1 class=\\"h4\\">Choppa's Home Page</h1>
    <div>
      <router-link-stub to=\\"/tribe/1/\\" tag=\\"a\\" ariacurrentvalue=\\"page\\" event=\\"click\\">View Tribe Page</router-link-stub> |
      <router-link-stub to=\\"/tribe/1/edit\\" tag=\\"a\\" ariacurrentvalue=\\"page\\" event=\\"click\\">Edit Tribe Page</router-link-stub> |
      <router-link-stub to=\\"/not-found-404-test\\" tag=\\"a\\" ariacurrentvalue=\\"page\\" event=\\"click\\">404</router-link-stub>
    </div>
  </div>
</div>"
`;
